### PR TITLE
feat: added experimental fields to conda lock

### PR DESCRIPTION
**Pull request by tdejager**
So that it is possible to re-create a package record from a locked package

### CODE REVIEW

**Review by baszalmstra — COMMENTED** [review]
Looks good to me! But could you also implement `From<RepoDataRecord>` and vice versa for `LockedPackage`? Then we are sure it will work!

File: crates/rattler_conda_types/src/conda_lock/mod.rs [hunk 1]
```diff
@@ -263,6 +264,52 @@ pub struct LockedDependency {
     pub source: Option<Url>,
     /// Build string
     pub build: Option<String>,
+
+    /// Experimental: architecture field
+    #[serde(skip_serializing_if = "Option::is_none")]
```
[the remark this answers is not in the document]
Comment (tdejager) [reply crates/rattler_conda_types/src/conda_lock/mod.rs:269]: Nice!

**Comment by tdejager** [comment]
I think I'm unable to implement From directly seeing as some fields can be `None` but not are not `None` in LockedPackage. For example the hashes are both options in `PackageRecord` but is an enum in `LockedPackage`. So I propose just to add some conversion methods, that return a Result for this.

**Comment by baszalmstra** [comment]
You could also implement TryFrom where the conversion could fail

**Comment by tdejager** [comment]
Did all that please check it again :) @baszalmstra

**Pull request merged**